samtranslator/model/api/http_api_generator.py

```diff
@@ -88,6 +102,71 @@ def _construct_http_api(self):
 
         return http_api
 
+    def _add_cors(self):
+        """
+        Add CORS configuration if CORSConfiguration property is set in SAM.
+        Adds CORS configuration only if DefinitionBody is present and
+        APIGW extension for CORS is not present in the DefinitionBody
+        """
+
+        if not self.cors_configuration:
+            return
+
+        if self.cors_configuration and not self.definition_body:
+            raise InvalidResourceException(
+                self.logical_id, "Cors works only with inline OpenApi specified in " "'DefinitionBody' property"
+            )
+
+        # If cors configuration is set to true add * to the allow origins
+        if isinstance(self.cors_configuration, bool):
+            # if cors config is true add Origins as "'*'"
+            properties = CorsProperties()
+
+        elif is_instrinsic(self.cors_configuration):
+            # Just set Origin property. Others will be defaults
+            properties = CorsProperties(AllowOrigins=self.cors_configuration)
+
+        elif isinstance(self.cors_configuration, dict):
+            # Make sure keys in the dict are recognized
+            if not all(key in CorsProperties._fields for key in self.cors_configuration.keys()):
+                raise InvalidResourceException(self.logical_id, "Invalid value for 'Cors' property")
+
+            properties = CorsProperties(**self.cors_configuration)
+
+        else:
+            raise InvalidResourceException(self.logical_id, "Invalid value for 'Cors' property")
+
+        if not OpenApiEditor.is_valid(self.definition_body):
+            raise InvalidResourceException(
+                self.logical_id,
+                "Unable to add Cors configuration because "
+                "'DefinitionBody' does not contain a valid "
+                "openapi definition",
```

Comment:
`OpenApi`

Also, add periods at the end of the error messages.

Reply:
Thank you, I have updated all the error messages to end with a period